libs/cardano-ledger-binary/test/Test/Cardano/Ledger/Binary/Vintage/Failure.hs

```diff
@@ -80,6 +93,12 @@ prop_shouldFailNegativeNatural = property $ do
   n <- forAll genNegativeInteger
   assertIsLeft (decode byronProtVer (toCBOR n) :: Either DecoderError Natural)
 
+prop_shouldFailMapWithDupKeys :: Property
+prop_shouldFailMapWithDupKeys = property $ do
+  xs <- forAll genDuplicateAssocList
+  let encodedMap = encodeMapLen (fromIntegral $ length xs) <> mconcat (toCBOR <$> xs)
```

Comment:
I am not sure what's the name of function in Hedgehog, but something an equivalent to `elements` from QuickCheck can be used to switch between indefinite and exact length encoding:
```suggestion
  encodedMap <- 
    elements [ encodeMapLen (fromIntegral $ length xs) <> foldMap toCBOR xs
             , encodeMapLenIndef <> foldMap toCBOR xs <> encodeBreak ]
```